src/lib/validate/validateArgs.js

```diff
@@ -0,0 +1,285 @@
+  const policy = parseCanaryPolicy(args, logger);
+  if (_.isEmpty(policy)) {
+    throw new Error(`Failed to parse the canary policy. Please double-check the configuration.`);
+
+  }
+  return policy;
+}
+
+async function validateArgs(inputs, args, logger, fcHelper) {
+  const {service, baseVersion} = args;
+  let baseVersionArgs = baseVersion;
+  if (baseVersionArgs != undefined) {
+    baseVersionArgs = baseVersionArgs.toString();
+  }
+  // check whether service in the plugin args is equal to the service deployed.
+  let curService;
+  if (service != undefined) {
+    const serviceDeployed =
+      inputs && inputs.props && inputs.props.service && inputs.props.service.name;
+    if (serviceDeployed !== service) {
+      throw new Error(
+        `The service names in args [${service}] and inputs [${serviceDeployed}] are different.`,
+      );
+    }
+    curService = service;
+  } else {
+    curService = _.get(inputs, 'props.service.name');
+  }
+
+  if (baseVersionArgs != undefined) {
+    await validateBaseVersion(curService, baseVersionArgs, fcHelper, logger);
+  }
+}
+
+/**
+ * Canary policy, only one of the following parameters can be selected, if not specified then no canary policy
+ * @param args
+ * @param logger
+ * @returns {{}}
+ */
+function parseCanaryPolicy(args, logger) {
+  let response = {};
+  const argsKeys = Object.keys(args);
+  logger.debug(`keys in args: [${argsKeys}].`);
+
+  let policies = [];
+  if (argsKeys.includes('canaryStep')) {
+    policies.push('canaryStep');
+  }
+  if (argsKeys.includes('canaryWeight')) {
+    policies.push('canaryWeight');
+  }
+  if (argsKeys.includes('canaryPlans')) {
+    policies.push('canaryPlans');
+  }
+  if (argsKeys.includes('linearStep')) {
+    policies.push('linearStep');
+  }
+  logger.debug(`User input canary policy: ${JSON.stringify(policies)}.`);
+
+  if (policies.length === 0) {
+    if (args.baseVersion !== null) {
+      logger.warn(`No canary policy found, the system will perform a full release.`);
+    }
+
+    response.key = 'full';
+    response.value = 100;
+  } else if (policies.length > 1) {
+    throw new Error(
+      `Only one canary policy can be selected, but [${policies.length}] canary policies are found.`,
+    );
+  } else {
+    // begin validate the strategy
+    const canaryPolicyName = policies[0];
+    // only canaryPlans input is an array.
+    logger.info(`Canary Policy: [${canaryPolicyName}].`);
+
+    if (canaryPolicyName === 'canaryPlans') {
+      // each plan in canaryPlans can't have a weight > 100
+      const plans = _.get(args, 'canaryPlans');
+      logger.debug(`canaryPlans: [${plans}].`);
+
+      if (plans == undefined) {
+        throw new Error(
+          `Format error, missing configuration of plan in canaryPlans, please check the configuration.`,
+        );
+      }
+
+      for (const plan of _.get(args, 'canaryPlans')) {
+        logger.debug(`plan: ${JSON.stringify(plan, null, 2)}.`);
+        if (plan.weight == undefined) {
```

Comment:
类型判断不用各个条件都判断一遍，直接采用assert会非常方便
```suggestion
        assert(plan.weight > 0 && plan.weight < 100, 'weight must be a number and  between 1 and 100')
```

Reply:
学习到了

Reply:
fixed